docs/developer/best_practices/commit.rst

```diff
@@ -0,0 +1,372 @@
+
+Example (Pull request `#1620 <https://github.com/scality/metalk8s/issues/1620>`_):
+
+- one commit for the cosmetic changes: `766f572e462c6933c8168a629ed4f479bb68a803 <https://github.com/scality/metalk8s/commit/766f572e462c6933c8168a629ed4f479bb68a803>`_
+
+- one commit for the functional changes: `3367fabdefc0b35d34bf7cf2fb0d33ff81f9fd5a <https://github.com/scality/metalk8s/commit/3367fabdefc0b35d34bf7cf2fb0d33ff81f9fd5a>`_
+
+Ideally, purely cosmetic changes which inflate the number of changes in a PR
+significantly, should go in a separate PR
+
+Refactoring
+:::::::::::
+
+When introducing new features, you often have to add new helpers or refactor
+existing code. In such case, instead of having single commit with everything
+inside, you can either:
+
+1. first add a new helper: `29f49cbe9dfa <https://github.com/scality/metalk8s/commit/29f49cbe9dfa0b824c818d25d4a2f6965351e65d>`_
+
+2. then use it in new code: `7e47310a8f20 <https://github.com/scality/metalk8s/commit/7e47310a8f20fd49f0ad36707b20e6c2a53df638>`_
+
+Or:
+
+1. first add the new code: `5b2a6d5fa498 <https://github.com/scality/metalk8s/commit/5b2a6d5fa49815180a2effdd37cb58542e83b5a5>`_
+
+2. then refactor the now duplicated code: `ac08d0f53a83 <https://github.com/scality/metalk8s/commit/ac08d0f53a835a0b2bc61c1fe5b7317bf4d6550c>`_
+
+Mixing unrelated changes
+::::::::::::::::::::::::
+
+It is sometimes tempting to do small unrelated changes as you are working on
+something else in the same code area.
+Please refrain to do so, or at least do it in a dedicated commit.
+
+Mixing non-related changes into the same commit makes revert and cherry-pick
+harder (and understanding as well).
+
+The pull request `#1846 <https://github.com/scality/metalk8s/pull/1846>`_ is a good example. It tackles three issues at once: `#1830 <https://github.com/scality/metalk8s/issues/1830>`_
+and `#1831 <https://github.com/scality/metalk8s/issues/1831>`_ (because they are similar) and `#839 <https://github.com/scality/metalk8s/issues/839>`_ (because it was making the other
+changes easier), but it uses distincts commits for each issue.
+
+How to write a commit message
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Why do we need commit messages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After comments in the code, commit messages are the easiest way to find context
+for every single line of code: running ``git blame`` on a file will give you,
+for each line, the identifier of the last commit that changed the line.
+
+Unlike a comment in the code (which applies to a single line or file), a commit
+message applies to a logical change and thus can provide information on the
+design of the code and why the change was done. This makes commit messages a
+part of the code documentation and makes them helpful for other developers to
+understand your code.
+
+Last but not least: commit messages can also be used for automating tasks such
+as issue management.
+
+Note that it is important to have all the necessary information in the commit
```

Comment:
We could wrap the `note` in an `rst` note format since it is information you want the readers to retain always

For example ```..Note: ```

Reply:
`Note that` here is more an expression than really introducing a note (I could totally remove it and just start the sentence with `It is important…`), si I'll keep it like that.